social/backends/open_id.py

```diff
@@ -277,14 +280,44 @@ class OpenIdConnectAuth(BaseOAuth2):
 
     Currently only the code response type is supported.
     """
-    ID_TOKEN_ISSUER = None
-    DEFAULT_SCOPE = ['openid']
+    # Override OIDC_ENDPOINT in your subclass to enable autoconfig of OIDC
+    OIDC_ENDPOINT = None
+
+    DEFAULT_SCOPE = ['openid', 'profile', 'email']
     EXTRA_DATA = ['id_token', 'refresh_token', ('sub', 'id')]
-    # Set after access_token is retrieved
-    id_token = None
+    REDIRECT_STATE = False
+    ACCESS_TOKEN_METHOD = 'POST'
+    REVOKE_TOKEN_METHOD = 'GET'
+
+    # Will be autoconfigured
+    oidc_config = None
+    ID_TOKEN_ISSUER = None
+    JWKS_KEYS = None
+    ACCESS_TOKEN_URL = None
+    AUTHORIZATION_URL = None
+    REVOKE_TOKEN_URL = None
+    SIGNING_ALGS = None
+
+    def init_oidc_config(self):
+        if self.oidc_config is None and self.OIDC_ENDPOINT:
+            # Cache these settings by setting class variables
+            cls = self.__class__
+            cls.oidc_config = self.get_json(self.OIDC_ENDPOINT + '/.well-known/openid-configuration')
+            cls.ACCESS_TOKEN_URL = self.oidc_config['token_endpoint']
+            cls.AUTHORIZATION_URL = self.oidc_config['authorization_endpoint']
+            cls.REVOKE_TOKEN_URL = self.oidc_config['revocation_endpoint']
```

Comment:
This line raised an exception for me because I don't have `revocation_endpoint` in my openid config.

The exception happens after `cls.oidc_config` has already been set. So the next time this won't get run and you won't get an exception, but you do end up with empty values for `cls.USERINFO_URL` and all other config defined below this line.